jupyter_server/gateway/connections.py

```diff
@@ -47,6 +47,8 @@ async def connect(self):
             url_escape(self.kernel_id),
             "channels",
         )
+        if self.session_id:
+            ws_url += f"?session_id={self.session_id}"
```

Comment:
Should `session_id` be escaped with `escape_url`? Also a test case would be nice :)

Reply:
used url_escape + add a test

Reply:
Added a test, but looks like Windows tests are failing for unrelated reason?

Reply:
The windows tests were fixed in #1441

Reply:
Thanks @blink1073 I synced to main and Windows tests work now, but now I get some coverage error. https://github.com/jupyter-server/jupyter_server/actions/runs/9821413970/job/27117179741?pr=1440 any pointers? Thanks